Let managers reset user's password

iagirre opened the pull request:
References
==========
This is a backport of the issue https://github.com/consul/consul/issues/2500 (done first at Madrid's fork).
The original PR is: https://github.com/AyuntamientoMadrid/consul/pull/1370

Objectives
==========
Let the manager change the users' password in two ways:

Send an email to the registered email so that the user can be able to change the password by themselves.
Change the password using a form. The manager can write it in plain text or generate a random value.
When the password is saved, the manager can print it for the user.

Visual Changes
=======================
- The new submenu and the overview
![password01](https://user-images.githubusercontent.com/31625251/37767514-ac7c4d0c-2dca-11e8-9708-6f2475aea698.gif)

- Random password generator and show/hide password
![password02_2](https://user-images.githubusercontent.com/31625251/37462218-72cb32cc-2851-11e8-82ff-88a75c5e86af.gif)

- Password saved and print password page
![password03](https://user-images.githubusercontent.com/31625251/37462226-7cf3aacc-2851-11e8-847f-caaac0780f55.png)

- How password is printed
![password04](https://user-images.githubusercontent.com/31625251/37462509-7d6d277a-2852-11e8-8763-28a855a75461.png)

Notes
=====================
I know that the 'Change user' link is still unchanged, but there is another PR waiting for merge that modifies that link. When both PR are merged the link will appear red and in the bottom of the box (I said this in order to avoid conflics when both PR will be merged, but If I should also change the link, let me know and I will).

bertocq commented:
@iagirre can you fix the failing spec at /spec/features/management/account_spec.rb:15 ? The solution in raul's PR will do fine. So we can merge this one straight away without waiting any longer for his PR to be completed

iagirre commented:
I fixed the spec that was failing: https://github.com/consul/consul/pull/2548/commits/d84895b800dbe24792c40f4e4d5517a9a6507905#diff-7f5afe0b6447090de432c81501a61459R9

Also, I changed the name because it was confusing. It was saying that the unverified users couldn't create spending proposals, but the whole spec was about editing their account, so I changed to give it a more descriptive name.

voodoorai2000 commented:
Excellent @iagirre 👌
